build_tools/benchmarks/run_benchmarks_on_android.py

```diff
@@ -373,11 +375,11 @@ def run_benchmarks_for_category(
       if do_capture and benchmark_key not in skip_captures:
         run_cmd = [
             "TRACY_NO_EXIT=1",
-            f"IREE_PRESERVE_DYLIB_TEMP_FILES={ANDROID_TMP_DIR}", "taskset",
+            f"IREE_PRESERVE_DYLIB_TEMP_FILES={ANDROID_TMP_DIR}",
+            "taskset",
             benchmark_info.deduce_taskset(),
-            os.path.join(ANDROID_TMP_DIR, TRACED_TOOL_REL_DIR,
-                         tool), f"--flagfile={MODEL_FLAGFILE_NAME}"
-        ]
+            os.path.join(ANDROID_TMP_DIR, TRACED_TOOL_REL_DIR, tool),
+        ] + tool_flags
```

Comment:
For trace captures we don't need to add `benchmark_format=json`. Actually the logic at L399 is written to try to match against the normal output.

Reply:
wow, thanks for the great bug catching here, @antiagainst .  I hadn't noticed the problem because I was only running pure trace-capture only runs, so `normal_benchmark_tool_dir` was empty and the faulty code wasn't exercised.